web/app/services/session.ts

```diff
@@ -16,6 +22,79 @@ export function isJSON(str: string) {
 
 export default class SessionService extends EmberSimpleAuthSessionService {
   @service declare router: RouterService;
+  @service declare fetch: FetchService;
+  @service declare session: SessionService;
+  @service declare flashMessages: FlashMessageService;
+
+  /**
+   * Whether the current session is valid.
+   * Set false if our poll response is 401, and when the
+   * user requires authentication with EmberSimpleAuth.
+   */
+  @tracked tokenIsValid = true;
+
+  /**
+   * Whether the service should show a reauthentication message.
+   * True when the user has dismissed a previous re-auth message.
+   */
+  @tracked preventReauthenticationMessage = false;
+
+
+  /**
+   * Whether the last poll response was a 401.
+   * Updated by the fetch service on every pollCall.
+   */
+  @tracked pollResponseIs401 = false;
+
+  /**
+   * A persistent task that periodically checks if the user's
+   * session has expired, and shows a flash message if it has.
+   * Kicked off by the Authenticated route.
+   */
+  pollForExpiredAuth = keepLatestTask(async () => {
+    await simpleTimeout(Ember.testing ? 100 : 10000);
+
+    this.fetch.fetch(
+      "/api/v1/me",
+      {
+        method: "HEAD",
+        headers: {
+          "Hermes-Google-Access-Token": this.data.authenticated.access_token,
+        },
```

Comment:
Not strictly necessary because the fetch service adds this automatically for backend requests.